tests/integration/targets/ndo_ipsla_monitoring_policy/tasks/main.yml

```diff
@@ -0,0 +1,434 @@
+  block:
+    
+    - name: Ensure sites exists
+      cisco.mso.mso_site:
+        <<: *mso_info
+        site: '{{ item.site }}'
+        apic_username: '{{ apic_username }}'
+        apic_password: '{{ apic_password }}'
+        apic_site_id: '{{ item.apic_site_id }}'
+        urls:
+          - https://{{ apic_hostname }}
+        state: present
+      loop:
+        - {site: "ansible_test", apic_site_id: 101}
+        - {site: "ansible_test_2", apic_site_id: 102}
+    
+    - name: Ensure tenant exist
+      cisco.mso.mso_tenant: 
+        <<: *mso_info
+        tenant: '{{ item }}'
+        users:
+          - '{{ mso_username }}'
+        sites:
+          - '{{ mso_site | default("ansible_test") }}'
+          - ansible_test_2
+        state: present
+      loop:
+        - ansible_test
+    
+    - name: Ensure templates do not exist
+      cisco.mso.ndo_template: &template_absent
+        <<: *mso_info
+        name: ansible_tenant_template
+        template_type: tenant
+        tenant: ansible_test
+        state: absent
+
+    - name: Ensure schemas do not exist
+      cisco.mso.mso_schema: &remove_schemas
+        <<: *mso_info
+        schema: '{{ mso_schema | default("ansible_test") }}'
+        state: absent
+    
+    - name: Ensure schema with template exists
+      cisco.mso.mso_schema_template:
+        <<: *mso_info
+        schema: '{{ mso_schema | default("ansible_test") }}'
+        tenant: ansible_test
+        template: ansible_test
+        state: present
+
+    - name: Ensure templates exist
+      cisco.mso.ndo_template:
+        <<: *template_absent
+        state: present
+
+    # CREATE
+
+    - name: Create a new IPSLA monitoring policy (check_mode)
+      cisco.mso.ndo_ipsla_monitoring_policy: &create_ipsla_monitoring_policy
+        <<: *mso_info
+        template: ansible_tenant_template
+        ipsla_monitoring_policy: ansible_test_ipsla_monitoring_policy
+        sla_frequency: 60
+        state: present
+      check_mode: true
+      register: cm_create_new_ipsla_monitoring_policy
+    
+    - name: Create a new IPSLA monitoring policy
+      cisco.mso.ndo_ipsla_monitoring_policy:
+        <<: *create_ipsla_monitoring_policy
+      register: nm_create_new_ipsla_monitoring_policy
+    
+    - name: Create a new IPSLA monitoring policy again
+      cisco.mso.ndo_ipsla_monitoring_policy:
+        <<: *create_ipsla_monitoring_policy
+      register: nm_create_new_ipsla_monitoring_policy_again
+
+    - name: Assert IPSLA monitoring policy was created
+      assert:
+        that:
+          - cm_create_new_ipsla_monitoring_policy is changed
+          - cm_create_new_ipsla_monitoring_policy.previous == {}
+          - cm_create_new_ipsla_monitoring_policy.current.name == "ansible_test_ipsla_monitoring_policy"
+          - cm_create_new_ipsla_monitoring_policy.current.slaType == "icmp"
+          - cm_create_new_ipsla_monitoring_policy.current.slaFrequency == 60
```

Comment:
if defaults should we assert those here?

Reply:
Removed the defaults